### PR TITLE
updates for identification of spec, schema

### LINKED ISSUE #561: Remove swagger: 2.0 from spec
The current OpenAPI spec has `"swagger": "2.0"` in it.  Since it's no longer called the swagger spec, this should go away.

I do believe we should continue to declare the version in the specification though, to make tooling more tolerant.  Some points about that.

* One major issue with the Swagger 2.0 tooling was that it was looking for the "2.0" version in the spec itself.  That meant, a "2.0.1" spec--even if structurally compatible--would not work.  We should advise all tooling vendors as to versioning process and how they should detect versions inside the spec.  Reference the concerns in #107.  In a couple cases we probably should have bumped the spec version in 2.0 to 2.0.1, etc.  Let's not make that mistake again

* Declaring a semver approach for the spec makes sense.  We don't need to go too far into the details in this ticket as to what constitutes a major, minor, patch version, but in general, if we put a version in the next spec version (which I advocate), it should be in the `x.y.z` format.

* Making it "2.0" as a string vs. number was dumb.  The above point solves that.

### PR DESCRIPTION
**Pull request by fehguy**
Fixes #561

### CODE REVIEW

**Comment by fehguy** [comment]
Also fixes #496

File: versions/3.0.md [hunk 1]
```diff
@@ -71,7 +71,7 @@ The schema exposes two types of fields. Fixed fields, which have a declared name
 
 The OAS representation of the API is made of a single file. However, parts of the definitions can be split into separate files, at the discretion of the user. This is applicable for `$ref` fields in the specification as follows from the [JSON Schema](http://json-schema.org) definitions.
 
-By convention, the OpenAPI Specification (OAS) file is named `swagger.json`.
+By convention, the OpenAPI Specification (OAS) file is named `oas.json`.
```
Comment (fehguy) [inline versions/3.0.md]: or oas.yaml???

Comment (earth2marsh) [inline versions/3.0.md]: or `openapi.yaml` and `openapi.json`

Comment (jasonh-n-austin) [inline versions/3.0.md]: +1

Comment (darrelmiller) [inline versions/3.0.md]: I prefer openapi.* if we must suggest something, although I prefer if we didn't express a preference for a filename.

File: versions/3.0.md [hunk 2]
```diff
@@ -106,7 +106,8 @@ This is the root document object for the API specification. It combines what pre
 
 Field Name | Type | Description
 ---|:---:|---
-<a name="oasSwagger"></a>swagger | `string` | **Required.** Specifies the OpenAPI Specification version being used. It can be used by the Swagger UI and other clients to interpret the API listing. The value MUST be `"2.0"`.
+<a name="oasSwagger"></a>swagger | `string` | **Required.** Specifies the OpenAPI Specification version being used. It can be used by tooling vendors and clients to interpret the version.  The structure shall be `major`.`minor`.`patch`, where `patch` versions _must_ be compatible with the existing `major`.`minor` tooling. Typically patch versions will be introduced to address errors in the documentation, and tooling should typically be compatible with the corresponding `major`.`minor` (3.0.*). Patch versions will correspond to patches of this document.
```
Comment (earth2marsh) [inline versions/3.0.md]: It looks like this is still called `swagger`, no? 

In the past, one challenge with versions is distinguishing the version of the spec vs the version of the API described by the spec.

Comment (fehguy) [inline versions/3.0.md]: my mistake, will fix this.

Comment (earth2marsh) [inline versions/3.0.md]: Another minor question, but changing if the anchor is being changed, is there any benefit in retaining the word `swagger`?

**Comment by fehguy** [comment]
Schema version + spec version will _not_ necessarily be the same.

@OAI/tdc to review by Weds and provide comments?

File: versions/3.0.md [hunk 3]
```diff
@@ -106,7 +106,8 @@ This is the root document object for the API specification. It combines what pre
...
+<a name="oasSchema"></a>schema | `string` | An optional URL to a formal OAS schema. If present, tooling _may_ choose to use this for validating the document, and the location _must_ be a formally approved OAS schema produced by the OpenAPI working group.
```
Comment (fehguy) [inline versions/3.0.md]: Kill this, will cause more harm than good @OAI/tdc 

Will publish a formal location of the schema (https://openapis.org/schemas/3.0-latest.json)

File: versions/3.0.md (hunk 1 above)
Comment (webron) [inline versions/3.0.md]: Another vote for `openapi.*`.

**Comment by webron** [comment]
Regarding the version - reiterating what we discussed (for documentation sake) - the benefits of a patch-level version granularity are clear, but we're also not going to be update the spec (normally) for every minor change. We'll do our best to collect changes and not have a too-frequent release cycle.

**Comment by webron** [comment]
Other than that, :+1: from me for the PR (assuming suggested file name and removal of `schema`).

**Comment by fehguy** [comment]
I have removed the schema reference and `swagger` from the key name.

**Comment by fehguy** [comment]
@OAI/tdc please approve or reject

**Comment by darrelmiller** [comment]
👍

**Comment by jasonh-n-austin** [comment]
:+1:

**Comment by earth2marsh** [comment]
👍

**Pull request merged**